### PR TITLE
Fix callstack view when switch between sampling and selection tabs

### PR DESCRIPTION
**Pull request by IrinaShkviro**
CallstackDataView was common for both `sampling` and `selection` tabs,
and after switching the tabs CallstackDataView showed the callstack from
the previously selected tab. This commit introduces new DataViewType for
`selection` callstack, and adds corresponding view for it.

Test: take a capture, make a selection, switch from selection to samping
tab and vice versa.

Bug: http://b/165613158

### CODE REVIEW

**Review by florian-kuebler — APPROVED** [review]
LGTM, but I guess you should wait for an additional approval, as I am not that familiar with that part of the code.

File: OrbitGl/CallStackDataView.cpp
Comment (reichlfl) [inline OrbitGl/CallStackDataView.cpp:16]: I'm not too happy with this change, even though I get that it's consistent. I'm fine with leaving it like that since it's more of a conceptual problem with the DataViewFactory, but let me put my thoughts down here nevertheless.

If I have a factory, I expect it to create objects of different types and behaviors (if only slighty) when I pass in different enum value. So when I look at this, I expect a different type of object being constructed for kCallstack and kSelectionCallstack, or at least an object that is initialized in a different way, resulting in - even if slightly - different behavior. This is not the case the Callstack - here it is basically used to create two instances of the same type. For the CallStackDataView, it is completely irrelevant that it is of type kSelectionCallStack. 

As I said, bit of a higher-level conceptual problem - to me, it does not make sense to have a factory with a "GetOrCreate" method. A factory's functionality shouldn't be to manage singleton-like objects. 

In the end, this factory interface is implemented exactly once, in OrbitApp. It could very well be replaced with `OrbitApp::GetOrCreateProcessesView()`, `OrbitApp::GetOrCreateModulesView()`. Or with a `OrbitDataViewCollection` class that does this as a member of OrbitApp, if we want to keep the modularization.

Edit: I've just seen that not all of the DataViewTypes are even supported by the factory... all the more reason to maybe remove it alltogether?

File: OrbitGl/SamplingReportDataView.cpp
Comment (reichlfl) [inline OrbitGl/SamplingReportDataView.cpp:227]: Another reason to strengthen my point above :-)

**Review by reichlfl — APPROVED** [review]
LGTM, even though I have some concerns with the factory in general as I've stated below. Removing the factory would make the data view management easier to read and understand, so we may want to tackle this at some point, but it's not super critical I guess.

**Comment by IrinaShkviro** [comment]
> LGTM, even though I have some concerns with the factory in general as I've stated below. Removing the factory would make the data view management easier to read and understand, so we may want to tackle this at some point, but it's not super critical I guess.

Thanks a lot for the suggestion! As discussed, I created a separate method for selection_callstack_data_view_, and used the same type as for callstack_data_view_. Please take another look

File: OrbitGl/App.h
Comment (reichlfl) [inline OrbitGl/App.h:204]: [[nodiscard]]

Comment (reichlfl) [reply OrbitGl/App.h:204]: Maybe add it for the method above as well since you're at it

**Review by reichlfl — APPROVED** [review]
LGTM, thanks for the change!

File: OrbitGl/App.h
Comment (IrinaShkviro) [reply OrbitGl/App.h]: Yes, thanks! Fixed it

**Pull request merged**
